main/acle.md

```diff
@@ -9382,6 +9415,2344 @@ are named after. All of the functions have external linkage.
+  __attribute__((arm_streaming))
+  void svstnt1_vnum[_u64_x4](svcount_t png, uint64_t *rn, int64_t vnum,
+                             svuint64x4_t zt);
+  
+  ```
+
+#### MOVT
+
+Move 8 bytes between general-purpose register and ZT0
+
+``` c
+  
+  __attribute__((arm_streaming_compatible, arm_shared_za, arm_preserves_za))
+  uint64_t svread_lane_zt(uint64_t zt, uint64_t imm_x8_offset);
+  
+  
+  __attribute__((arm_streaming_compatible, arm_shared_za, arm_preserves_za))
+  void svwrite_lane_zt(uint64_t zt, uint64_t imm_x8_offset, uint64_t rt);
+  
+  ```
+
+#### LDR, STR
+
+Spill and fill of ZT0
+
+``` c
+  
+  __attribute__((arm_streaming_compatible, arm_shared_za, arm_preserves_za))
+  void svldr_zt(uint64_t zt, const void *rn);
+  
+  
+  __attribute__((arm_streaming_compatible, arm_shared_za, arm_preserves_za))
+  void svstr_zt(uint64_t zt, void *rn);
+  
+  ```
+
+#### ZERO
+
+Zero ZT0
+
+``` c
+  
+  __attribute__((arm_streaming_compatible, arm_shared_za, arm_preserves_za))
+  void svzero_zt(uint64_t zt);
+  
+  ```
+
+#### LUTI2, LUTI4
+
+Lookup table read with 2-bit and 4-bit indexes
+
+``` c
+  // Variants are also available for _zt[_u16] and _zt[_u32]
+  __attribute__((arm_streaming, arm_shared_za, arm_preserves_za))
+  svuint8_t svluti2_lane_zt[_u8](uint64_t zt, svuint8_t zn, uint64_t imm_idx);
+  
+  
+  // Variants are also available for _zt[_u16] and _zt[_u32]
+  __attribute__((arm_streaming, arm_shared_za, arm_preserves_za))
+  svuint8x2_t svluti2_lane_zt[_u8]_x2(uint64_t zt, svuint8_t zn,
+                                      uint64_t imm_idx);
+  
+  
+  // Variants are also available for _zt[_u16] and _zt[_u32]
+  __attribute__((arm_streaming, arm_shared_za, arm_preserves_za))
+  svuint8x4_t svluti2_lane_zt[_u8]_x4(uint64_t zt, svuint8_t zn,
+                                      uint64_t imm_idx);
+  
+  
+  // Variants are also available for _zt[_u16] and _zt[_u32]
+  __attribute__((arm_streaming, arm_shared_za, arm_preserves_za))
+  svuint8_t svluti4_lane_zt[_u8](uint64_t zt, svuint8_t zn, uint64_t imm_idx);
+  
+  
+  // Variants are also available for _zt[_u16] and _zt[_u32]
+  __attribute__((arm_streaming, arm_shared_za, arm_preserves_za))
+  svuint8x2_t svluti4_lane_zt[_u8]_x2(uint64_t zt, svuint8_t zn,
+                                      uint64_t imm_idx);
+  
+  
+  // Variants are also available for _zt[_u32]
+  __attribute__((arm_streaming, arm_shared_za, arm_preserves_za))
+  svuint16x4_t svluti4_lane_zt[_u16]_x4(uint64_t zt, svuint16_t zn,
+                                        uint64_t imm_idx);
+  
+  ```
+
+#### MOVA
+
+Move multi-vectors to/from ZA
+
+``` c
+  // Variants are also available for _za8[_u8], _za16[_s16], _za16[_u16],
+  // _za16[_f16], _za16[_bf16], _za32[_s32], _za32[_u32], _za32[_f32],
+  // _za64[_s64], _za64[_u64] and _za64_[f64]
+  __attribute__((arm_streaming, arm_shared_za, arm_preserves_za))
+  svint8x2_t svread_hor_za8[_s8]_vg2(uint32_t slice_base,
```

Comment:
I think for elements 16-bits and above (za16-za64) we should be passing in a tile index for both the reads and writes. It seems to be only the 8-bit versions that are restricted to tile 0 only.

Reply:
Thanks, I've addressed that suggestion in https://github.com/ARM-software/acle/pull/217/commits/d7d716d65de5661ed730577d8c9b76a0bccb534c